spark/src/main/java/org/apache/iceberg/actions/RemoveOrphanFilesAction.java

```diff
@@ -254,4 +270,53 @@ private static void listDirRecursively(
       return files.iterator();
     };
   }
+
+  protected static List<String> findOrphanFiles(
+      Dataset<Row> validFileDF,
+      Dataset<Row> actualFileDF) {
+    Column nameEqual = filename.apply(actualFileDF.col(RELATIVE_FILE_PATH))
+        .equalTo(filename.apply(validFileDF.col(RELATIVE_FILE_PATH)));
+
+    Column pathContains = actualFileDF.col(RELATIVE_FILE_PATH)
+        .contains(validFileDF.col(RELATIVE_FILE_PATH));
+
+    Column joinCond = nameEqual.and(pathContains);
```

Comment:
Isn't PathContains always true if nameEqual? I feel like you just need nameEqual Here

Reply:
@RussellSpitzer `nameEquals` is only for fileName and does not include path. Also this condition was already there so didnt want to change it.